lib/base/setup-device.js

```diff
@@ -19,7 +19,8 @@ const async = require('async'),
     }
 
     function isValidDeviceName(name) {
-        return ['$', '%'].indexOf(name[0]) === -1;
+        const re = new RegExp("^[_a-zA-Z][a-zA-Z0-9#_-]*");
+        return (name === String(name.match(re)));
```

Comment:
already name is string, so name.match(re) is enough as like #27

Reply:
name.match(re) is object.